src/controllers/krate/publish.rs

```diff
@@ -446,11 +471,9 @@ pub async fn publish(app: AppState, req: BytesRequest) -> AppResult<Json<GoodCra
 
             SendPublishNotificationsJob::new(version.id).enqueue(conn)?;
 
-            // If this is a new version for an existing crate it is sufficient
-            // to update the default version asynchronously in a background job.
-            if inserted_default_versions == 0 {
-                UpdateDefaultVersion::new(krate.id).enqueue(conn)?;
-            }
+            // Update the default version asynchronously in a background job
+            // to ensure correctness and eventual consistency.
+            UpdateDefaultVersion::new(krate.id).enqueue(conn)?;
```

Comment:
I guess we could move this into the `if let Some(existing_default_version) = existing_default_version` condition above. we don't need to enqueue it in the `insert_into()` case.

Reply:
So rather than always enqueuing the update job, we only enqueue when it's not a new crate, namely when releasing a new version, right? 

Seems right to me.